### DISCUSSION TITLE
Error invoking platform.describe()

### DISCUSSION THREAD

**Discussion by msdevega** [Pi4J/pi4j#187] [Q&A] [answered]
I am trying to run my first pi4j-v2 program in my raspberry 4B with this code:

`
import com.pi4j.Pi4J;
import com.pi4j.io.gpio.digital.DigitalOutput;
import com.pi4j.platform.Platform;
import com.pi4j.platform.Platforms;
import com.pi4j.provider.Providers;
import com.pi4j.registry.Registry;
import com.pi4j.util.Console;

public class GettingStartedExample {

    public static void main(String[] args) throws Exception {

        final var console = new Console();

        console.title("<-- The Pi4J Project -->", "Getting Started Example");

        var pi4j = Pi4J.newAutoContext();

        Platforms platforms = pi4j.platforms();

        console.box("Pi4J PLATFORMS");
        platforms.describe().print(System.out);

        console.println();

        Platform platform = pi4j.platform();

        console.box("Pi4J DEFAULT PLATFORM");
// null pointer exception here. 
        platform.describe().print(System.out);

        console.println();

        Providers providers = pi4j.providers();
        console.box("Pi4J PROVIDERS");
        providers.describe().print(System.out);
        console.println();

        Registry registry = pi4j.registry();

        DigitalOutput output = pi4j.dout().create(1, "my-digital-output-1");

        console.box("Pi4J REGISTRY");
        registry.describe().print(System.out);
        console.println();

        pi4j.shutdown();
    }
}`

the code compile and execute well, but it seems that the Platform object is not created and we get a null pointer exception when
call the describe() function:
`Exception in thread "main" java.lang.NullPointerException: Cannot invoke "com.pi4j.platform.Platform.describe()" because "platform" is null
	at simple.GettingStartedExample.main(GettingStartedExample.java:117)`

This the full output:
`ene 11, 2022 5:03:02 P.?M. com.pi4j.util.Console print
INFO: 

ene 11, 2022 5:03:02 P.?M. com.pi4j.util.Console println
INFO: ************************************************************
ene 11, 2022 5:03:02 P.?M. com.pi4j.util.Console println
INFO: ************************************************************
ene 11, 2022 5:03:02 P.?M. com.pi4j.util.Console println
INFO: 
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO:                   <-- The Pi4J Project -->                  
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO:                    Getting Started Example                  
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: 
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: ************************************************************
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: ************************************************************
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: 
ene 11, 2022 5:03:03 P.?M. com.pi4j.Pi4J newAutoContext
INFO: New auto context
ene 11, 2022 5:03:03 P.?M. com.pi4j.Pi4J newContextBuilder
INFO: New context builder
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: --------------------
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: |  Pi4J PLATFORMS  |
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: --------------------
PLATFORMS: [0] "Pi4J Runtime Platforms" <com.pi4j.platform.impl.DefaultPlatforms> 
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: 
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: ---------------------------
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: |  Pi4J DEFAULT PLATFORM  |
ene 11, 2022 5:03:03 P.?M. com.pi4j.util.Console println
INFO: ---------------------------
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "com.pi4j.platform.Platform.describe()" because "platform" is null
	at simple.GettingStartedExample.main(GettingStartedExample.java:117)
`

Any idea of what is the problem and how to solve it?

best regrads

**Comment by FDelporte** [comment]
Is this a Maven project? Which Pi4J dependencies did you include?
As you can see in the minimal example pom.xml at https://github.com/Pi4J/pi4j-example-minimal/blob/main/pom.xml#L66 the raspberrypi platform plugin is needed.

**Comment by msdevega** [comment]
No, I am compiling directly with javac and used this jars:
`
pi4j-core-2.1.1.jar	
slf4j-api-1.7.30.jar	
slf4j-jdk14-1.7.9.jar
`

**Comment by msdevega** [comment, marked as answer]
I add the pi4j-plugin-raspberry-2.1.1.jar and all is working now. 
Thanks for the advice. 
I will read some maven tutorial to use the recommend maven development system.

**Reply by FDelporte** [reply]
Great!
If you want a Maven example, check: https://pi4j.com/getting-started/minimal-example-application/
Actually... we are missing a pure javac-getting-started page on the site. Can you maybe share your sources and a short description of how you compile and run? If so, please mail to frank at pi4j dot com.

**Comment by msdevega** [comment]
I can do this howto. I will ask frank about the format and where to upload it.
